static/main.js

```diff
@@ -32,9 +40,27 @@ function endPos() {
 	ctx.beginPath();
 
 	// send info using websocket
-	const snapshot = ctx.getImageData(0, 0, canvas.width, canvas.height);
-	console.log(snapshot);
-	socket.emit('send-stroke', snapshot);
+	const board_after = ctx.getImageData(0, 0, canvas.width, canvas.height);
+	console.log(board_after);
+
+	// get difference btw initial & after board
+	let coord = [];
+	let val = [];
+	for (let i = 0; i < board_after.data.length; i++) {
+	    if (board_initial.data[i] != board_after.data[i]) {
+	        coord.push(i);
+	        val.push(board_after.data[i]);
+	    }
+	}
+
+	// create JSON
+	var diff = {
+	    "coord" : coord,
+	    "val" : val
+	};
```

Comment:
We can encapsulate this into its own function that takes in the before and after and returns the dict.

Reply:
Encapsulated into its own function as getDiff() which returns diff as JSON